### ISSUE TITLE
npm run package fails on gulp copy --option noreload

### ISSUE THREAD

**Issue by vecernik** [Teradata/covalent-electron#51]
In latest master, npm run package on osx/node-8.4 produces following error:
$ npm run package

> covalent-electron@0.0.1 package /Volumes/Larder/Downloads/covalent-electron-develop
> npm run ngbuild && npm run build && npm run npm-install && node "./node_modules/electron-packager/cli.js" dist Covalent --icon=src/app/assets/ico/icon.icns --out=dist-app --overwrite

> covalent-electron@0.0.1 ngbuild /Volumes/Larder/Downloads/covalent-electron-develop
> rimraf ./dist && ng build

 11% building modules 13/18 modules 5 active ...ode_modules/style-loader/addStyles.js(node:23265) DeprecationWarning: loaderUtils.parseQuery() received a non-string value which can be problematic, see https://github.com/webpack/loader-utils/issues/56
parseQuery() will be replaced with getOptions() in the next major version of loader-utils.
Hash: e0a0529e5a70fbdd3ffb                                                               
Time: 10577ms
chunk    {0} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 962 kB {5} [initial] [rendered]
chunk    {1} main.bundle.js, main.bundle.js.map (main) 32.2 kB {4} [initial] [rendered]
chunk    {2} styles.bundle.js, styles.bundle.js.map (styles) 459 kB {5} [initial] [rendered]
chunk    {3} scripts.bundle.js, scripts.bundle.js.map (scripts) 109 kB {5} [initial] [rendered]
chunk    {4} vendor.bundle.js, vendor.bundle.js.map (vendor) 5.07 MB [initial] [rendered]
chunk    {5} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]

> covalent-electron@0.0.1 build /Volumes/Larder/Downloads/covalent-electron-develop
> gulp copy --option noreload

/Volumes/Larder/Downloads/covalent-electron-develop/node_modules/require-dir/index.js:97
            if (!require.extensions.hasOwnProperty(ext)) {
                                    ^

TypeError: require.extensions.hasOwnProperty is not a function
    at requireDir (/Volumes/Larder/Downloads/covalent-electron-develop/node_modules/require-dir/index.js:97:37)
    at Object.<anonymous> (/Volumes/Larder/Downloads/covalent-electron-develop/gulpfile.js:3:23)
    at Module._compile (module.js:573:30)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)
    at Function.Module._load (module.js:462:3)
    at Module.require (module.js:517:17)
    at require (internal/module.js:11:18)
    at Liftoff.handleArguments (/Volumes/Larder/Downloads/covalent-electron-develop/node_modules/gulp/bin/gulp.js:116:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! covalent-electron@0.0.1 build: `gulp copy --option noreload`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the covalent-electron@0.0.1 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/michal/.npm/_logs/2017-08-31T09_02_06_319Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! covalent-electron@0.0.1 package: `npm run ngbuild && npm run build && npm run npm-install && node "./node_modules/electron-packager/cli.js" dist Covalent --icon=src/app/assets/ico/icon.icns --out=dist-app --overwrite`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the covalent-electron@0.0.1 package script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

Is that a new issue or did I overlook something?

**Comment by renarded** [comment]
Hi, 

Well, I just happened to get this exact same error (running npm 5.3.0 and node 8.4.0)

The problem comes from the require-dir library, which is broken with node 8. It has been solved here : https://github.com/aseemk/requireDir/releases/tag/0.3.2 

So what I did is modify the package.json file : 

- replace  "require-dir": "0.3.0",   by "require-dir": "^0.3.2",
- rm -rf node_modules
- npm install
- npm run package

And it worked. I hope this will help you.

--------------
For the package maintainer : I also wonder why there is no ^ in front of most version dependencies in package.json. Is it because keeping the last version of all package will be likely to break things ?
